casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala

```diff
@@ -537,8 +537,13 @@ object BlockAPI {
   )(implicit casper: MultiParentCasper[F]): F[A] =
     for {
       dag <- casper.blockDag
-      normalizedFaultTolerance <- SafetyOracle[F]
-                                   .normalizedFaultTolerance(dag, block.blockHash) // TODO: Warn about parent block finalization
+      // TODO this is temporary solution to not calculate fault tolerance for old blocks which is costly
+      oldBlock = dag.latestBlockNumber.map(_ - block.body.state.blockNumber).map(_ > 100)
+      normalizedFaultTolerance <- oldBlock.ifM(
+                                   1f.pure[F],
+                                   SafetyOracle[F]
+                                     .normalizedFaultTolerance(dag, block.blockHash)
+                                 ) // TODO: Warn about parent block finalization
```

Comment:
@nzpr, @zsluedem do you know what this comment means?

Reply:
I cannot imagine what this might mean

Reply:
I don't know, either.